samza-core/src/test/java/org/apache/samza/operators/TestJoinOperator.java

```diff
@@ -243,13 +262,23 @@ private StreamOperatorTask createStreamOperatorTask(Clock clock) throws Exceptio
 
     Config config = mock(Config.class);
 
-    StreamApplication sgb = new TestStreamApplication();
-    StreamOperatorTask sot = new StreamOperatorTask(sgb, runner, clock);
+    StreamOperatorTask sot = new StreamOperatorTask(app, runner, clock);
     sot.init(config, taskContext);
     return sot;
   }
 
-  private static class TestStreamApplication implements StreamApplication {
+  private static class TestJoinStreamApplication implements StreamApplication {
+
+    private final TestJoinFunction joinFn;
+
+    TestJoinStreamApplication(TestJoinFunction joinFn) {
```

Comment:
Looks like we always take the same TestJoinFunction. Maybe take a TestJoinFunction in createStreamOperatorTask and remove the default constructor?